### DISCUSSION TITLE
Question regarding valid use of the mutation operation

### DISCUSSION THREAD

**Discussion by benhavo** [graphql/graphql-spec#1028] [Q&A] [answered]
The spec indicates that a mutation is meant to model the "write and then fetch" operation. However, I have a use case in which I need to perform an action on the server without writing data directly to the database. The operation should trigger a call to a third party API to send an email. Is it appropriate to use a mutation for this use case? If not, is another operation more appropriate? I can't imagine there exists a framework for API communication that doesn't allow for this scenario...

**Comment by benjie** [comment, marked as answer]
Yes, mutations should be used in any case where a side effect takes place. GraphQL does not relate to databases or any particular storage or business logic technology. It is indifferent to where the side effect (or "write") takes place, only that it should be triggered from a mutation field.

**Reply by benhavo** [reply]
Thank you for the clarification!
